docs/adr/0041-gcp-cloud-tasks-for-job-queuing.md

```diff
@@ -0,0 +1,80 @@
+---
+status: proposed
+date: 2024-01-10
+deciders: Ben Bangert, Dan Schomberg
+informed: FxA Team
+---
+
+# Use Google Cloud Tasks for job queueing
+
+## Context and Problem Statement
+
+Many account related tasks require interaction with multiple systems that may exceed the typical request deadline. Some of these tasks require additional guarantee's they fail, or may partially complete before erroring and require a retry. We also occasionally run into cases where there's many thousands or millions of tasks to complete that have been handled before by splitting up lists of arguments and running them in parallel across many nodes manually by our SRE's. Another challenge we've faced when running large quantities of tasks in a batch manner is ensuring rate limits or adverse system conditions aren't hit.
```

Comment:
> Some of these tasks require additional guarantee's they fail,

I think I understand what this is saying but I'm not sure.  So a rephrasing might help.

Reply:
I think it's missing the word 'when'